daemon/entrypoint.sh

```diff
@@ -110,24 +110,40 @@ function populate_kv {
 #######
 
 function start_mon {
-  if [ ! -n "$CEPH_PUBLIC_NETWORK" ]; then
+  if [[ ! -n "$CEPH_PUBLIC_NETWORK" && ${NETWORK_AUTO_DETECT} -eq 0 ]]; then
     echo "ERROR- CEPH_PUBLIC_NETWORK must be defined as the name of the network for the OSDs"
     exit 1
   fi
 
-  if [ ${MON_IP_AUTO_DETECT} -eq 1 ]; then
-    MON_IP=$(ip -6 -o a | grep scope.global | awk '/eth/ { sub ("/..", "", $4); print $4 }' | head -n1)
-    if [ -z "$MON_IP" ]; then
-      MON_IP=$(ip -4 -o a | awk '/eth/ { sub ("/..", "", $4); print $4 }')
+  if [[ ! -n "$MON_IP" && ${NETWORK_AUTO_DETECT} -eq 0 ]]; then
+    echo "ERROR- MON_IP must be defined as the IP address of the monitor"
+    exit 1
+  fi
+
+  if [ ${NETWORK_AUTO_DETECT} -ne 0 ]; then
+    if [ command -v ip ]; then
+      if [ ${NETWORK_AUTO_DETECT} -eq 1 ]; then
+        MON_IP=$(ip -6 -o a | grep scope.global | awk '/eth/ { sub ("/..", "", $4); print $4 }' | head -n1)
```

Comment:
`<insignificant>`
For clarity, it might be nice to write the `ip` command out:
`ip -6 -oneline addr show | ...`
`</insignificant>`